unity-client/Assets/Scripts/MainScripts/DCL/Components/Audio/DCLAudioStream.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections;
+using UnityEngine;
+
+namespace DCL.Components
+{
+    public class DCLAudioStream : BaseComponent
+    {
+        [System.Serializable]
+        public class Model
+        {
+            public string url;
+            public bool playing = false;
+            public float volume = 1;
+        }
+
+        public Model model;
+        private bool isPlaying = false;
+
+        public override IEnumerator ApplyChanges(string newJson)
+        {
+            Model prevModel = model;
+            model = SceneController.i.SafeFromJson<Model>(newJson);
+
+            bool forceUpdate = prevModel.volume != model.volume;
+
+            UpdatePlayingState(forceUpdate);
+
+            yield return null;
+        }
+
+        private void Start()
+        {
+            CommonScriptableObjects.playerCoords.OnChange += OnPlayerCoordsChanged;
+            CommonScriptableObjects.rendererState.OnChange += OnRendererStateChanged;
+        }
+
+        private void OnDestroy()
+        {
+            CommonScriptableObjects.playerCoords.OnChange -= OnPlayerCoordsChanged;
+            CommonScriptableObjects.rendererState.OnChange -= OnRendererStateChanged;
+            StopStreaming();
+        }
+
+        private bool AreCoordsInsideComponentScene(Vector2Int coords)
+        {
+            if (scene == null) return false;
+            return scene.parcels.Contains(coords);
+        }
+
+        private void UpdatePlayingState(bool forceStateUpdate)
+        {
+            if (gameObject.activeInHierarchy)
```

Comment:
Too much `if...else` and flags spaguetti here. There isn't a way to improve this logic? Is a bit hard to follow and will be a struggle to modify it in the future.

Reply:
I think the logic is pretty easy to follow. I would early-exit (`if (!gameObject.activeInHierarchy) return`) but other than that it's pretty easy to follow the next 4 ifs

Reply:
> I think the logic is pretty easy to follow. I would early-exit (`if (!gameObject.activeInHierarchy) return`) but other than that it's pretty easy to follow the next 4 ifs

I've have already made those changes. Waiting for CI tests and merge? :)